.github/workflows/build.yaml

```diff
@@ -262,6 +265,45 @@ jobs:
           echo ${{ secrets.GITHUB_TOKEN }} | helm registry login -u ${{ github.actor }} --password-stdin ${{ env.OCI_REGISTRY }}
           helm push ${{ env.ARTIFACT_DIR }}/${{ env.HELM_PACKAGE_DIR }}/radius-${{ env.CHART_VERSION }}.tgz oci://${{ env.OCI_REGISTRY }}/${{ env.OCI_REPOSITORY }}
 
+  build-and-push-bicep-types:
```

Comment:
I'd suggest refactoring this to move the "skip" logic down to the part where we publish to ACR. If you want to be really fancy you could use two separate jobs for it.

The reason is that we want to run as much of the workflow as possible on every commit. That way we can catch breaks as early as possible.

Reply:
Updated to only publish to ACR on main/releases. Everything else should run every time